game/engine/main.py

```
@@ -203,23 +208,30 @@ def valid_queen(fr, fc, tr, tc):
 def valid_king(color, fr, fc, tr, tc):
     if abs(tr - fr) <= 1 and abs(tc - fc) <= 1:
         return True
-        
+
     if fr == tr and abs(tc - fc) == 2:
         if color == 'white' and fr == 7 and fc == 4:
             if tc == 6 and W_K_CASTLE and is_empty(BOARD[7][5]) and is_empty(BOARD[7][6]):
-                if not is_square_attacked(7, 4, 'black') and not is_square_attacked(7, 5, 'black') and not is_square_attacked(7, 6, 'black'):
+                if (not is_square_attacked(7, 4, 'black') and
+                        not is_square_attacked(7, 5, 'black') and
+                        not is_square_attacked(7, 6, 'black')):
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)